[content-visibility] c-v:hidden only turns on contain: size, not contain: inline-size

cathiechen opened the pull request:
#### eaa26fbd160ee4fac5713f73783a6f582323e0ae
<pre>
[content-visibility] c-v:hidden only turns on contain: size, not contain: inline-size
<a href="https://bugs.webkit.org/show_bug.cgi?id=248280">https://bugs.webkit.org/show_bug.cgi?id=248280</a>

Reviewed by Rob Buis.

Content-visibility:hidden makes that element skips its contents. Per [1], when an element skips its contents,
it turns on contain:size, not contain:inline-size.

[1] <a href="https://w3c.github.io/csswg-drafts/css-contain-2/#skips-its-contents">https://w3c.github.io/csswg-drafts/css-contain-2/#skips-its-contents</a>

* LayoutTests/imported/w3c/web-platform-tests/css/css-sizing/contain-intrinsic-size/contain-intrinsic-size-033-expected.txt:
* Source/WebCore/rendering/RenderElement.h:
(WebCore::RenderElement::shouldApplyInlineSizeContainment const):

Canonical link: <a href="https://commits.webkit.org/257431@main">https://commits.webkit.org/257431@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/137cafea54e85d6b17bfb6a3dc06b87602a9cf28

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/98857 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/8068 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/32006 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/108272 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/168529 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/102812 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/8619 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/85435 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/91390 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/106259 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/104540 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/6530 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/90090 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/33563 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/88352 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/21433 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/76420 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/1979 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/22964 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/1888 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/60/builds/45429 "Found 1 new test failure: media/video-inactive-playback.html (failure)") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/5118 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/6844 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/42422 "Passed tests") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/3287 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

Loirooriol reviewed (approved):
Informal LGTM.

I was confused at first because I thought that `shouldApplyInlineSizeContainment()` was supposed to be true for `contain: size` (i.e. full size containment implies both inline-size and block-size containment).

But if I get it right, `containsInlineSize()` is only true for `contain: inline-size`, not for `contain: size`, and likewise `shouldApplyInlineSizeContainment()` should only be true when there is only inline-size containment. Then the patch is good.

BTW, test auto-009 should be resynced, these two failures that you are removing aren't actually supposed to pass after the last CSSWG resolutions, so I removed them and added different tests.

rwlbuis reviewed (approved):
LGTM.

cathiechen committed:
[content-visibility] c-v:hidden only turns on contain: size, not contain: inline-size
https://bugs.webkit.org/show_bug.cgi?id=248280

Reviewed by Rob Buis.

Content-visibility:hidden makes that element skips its contents. Per [1], when an element skips its contents,
it turns on contain:size, not contain:inline-size.

[1] https://w3c.github.io/csswg-drafts/css-contain-2/#skips-its-contents

* LayoutTests/imported/w3c/web-platform-tests/css/css-sizing/contain-intrinsic-size/contain-intrinsic-size-033-expected.txt:
* Source/WebCore/rendering/RenderElement.h:
(WebCore::RenderElement::shouldApplyInlineSizeContainment const):

Canonical link: https://commits.webkit.org/257431@main
